src/foundry/common/data/data.mjs/chatMessageData.d.ts

```diff
@@ -0,0 +1,207 @@
+import { FieldReturnType, PropertiesToSource } from '../../../../types/helperTypes';
+import { DocumentData } from '../../abstract/module.mjs';
+import { CONST, documents } from '../../module.mjs';
+import * as fields from '../fields.mjs';
+import { ChatSpeakerData, ChatSpeakerDataConstructorData } from './chatSpeakerData';
+
+interface ChatMessageDataSchema extends DocumentSchema {
+  _id: typeof fields.DOCUMENT_ID;
+  type: DocumentField<CONST.ChatMessageType> & {
+    type: typeof Number;
+    required: true;
+    default: typeof CONST.CHAT_MESSAGE_TYPES.OTHER;
+    validate: typeof _validateChatMessageType;
+    validationError: 'The provided ChatMessage type must be in CONST.CHAT_MESSAGE_TYPES';
+  };
+  user: fields.ForeignDocumentField<{ type: documents.BaseUser; required: true }>;
+  timestamp: FieldReturnType<typeof fields.TIMESTAMP_FIELD, { required: true }>;
+  flavor: typeof fields.STRING_FIELD;
+  content: typeof fields.BLANK_STRING;
+  speaker: DocumentField<typeof ChatSpeakerData> & { required: true; default: {} };
+  whisper: DocumentField<string[]> & {
+    type: typeof String[];
+    clean: (users: unknown[]) => string[];
+    required: true;
+    default: string[];
+  };
+  blind: typeof fields.BOOLEAN_FIELD;
+  roll: typeof fields.JSON_FIELD;
+  sound: typeof fields.AUDIO_FIELD;
+  emote: typeof fields.BOOLEAN_FIELD;
+  flags: typeof fields.OBJECT_FIELD;
+}
+
+interface ChatMessageDataProperties {
+  /**
+   * The _id which uniquely identifies this ChatMessage document
+   */
+  _id: string | null;
+
+  /**
+   * The message type from CONST.CHAT_MESSAGE_TYPES
+   * @defaultValue `CONST.CHAT_MESSAGE_TYPES.OTHER`
+   */
+  type: number;
```

Comment:
This and its ConstructorData counterpart should be `CONST.ChatMessageType` instead of `number`.